lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -177,6 +177,83 @@ static Value createSoftmaxBackwardCommonKernel(PatternRewriter &rewriter,
   return sub;
 }
 
+// Checks whether the `shapeA` and `shapeB` are broadcast compatible or not. If
+// yes, then computes the final broadcast shape.
+static bool computeBroadcastShape(PatternRewriter &rewriter, Location loc,
```

Comment:
We try to avoid copying and pasting code as much as possible to make maintenance easier. If you need this, you can move the implementation to `Dialect/Torch/Utils/Utils.{h,cpp}`, which is included by `DecomposeComplexOps.cpp` and `TorchToTMTensor.cpp` (the other file that uses this helper)